contracts/risk/CoverageDataProvider.sol

```diff
@@ -0,0 +1,406 @@
+    }
+
+    /***************************************
+     VIEW FUNCTIONS
+    ***************************************/
+   
+    /**
+     * @notice Returns the `SOLACE`.
+     * @return solace_ The address of the `SOLACE` token.
+    */
+    function solace() public view override returns (address solace_) {
+      return address(_solace);
+    }
+
+    /**
+     * @notice Returns the `SOLACE/USDC` SLP pool.
+     * @return solaceUsdcPool_ The address of the pool.
+    */
+    function solaceUsdcPool() public view override returns (address solaceUsdcPool_) {
+      return address(_solaceUsdcPool);
+    }
+
+    /**
+     * @notice Returns registry address.
+     * @return registry_ The registry address.
+    */
+    function registry() external view override returns (address registry_) {
+      return address(_registry);
+    }
+
+    /**
+     * @notice Returns the price oracle.
+     * @return oracle The address of the price oracle.
+    */
+    function priceOracle() public view override returns (address oracle) {
+      return address(_priceOracle);
+    }
+
+    /**
+     * @notice Returns status of the underwriting pool.
+     * @param pool_ The underwriting pool to query status.
+     * @return status True if underwriting pool is enabled. 
+    */
+    function poolStatus(address pool_) public view override returns (bool status) {
+      return _underwritingPoolStatus[pool_];
+    }
+
+    /**
+     * @notice Returns the underwriting pool for the given index.
+     * @param index_ The underwriting pool index.
+     * @return pool The underwriting pool.
+    */
+    function poolAt(uint256 index_) external view override returns (address pool) {
+      return _indexToUnderwritingPool[index_];
+    }
+
+    /**
+     * @notice Returns the underwriting pool count.
+     * @return count The number of underwriting pools.
+    */
+    function numOfPools() public view override returns (uint256 count) {
+      return _underwritingPoolCount;
+    }
+
+    /**
+     * @notice Returns the underwriting pool asset count.
+     * @return count The number of underwriting pools.
+    */
+    function numOfAssets() public view override returns (uint256 count) {
+      return _assetCount;
+    }
+
+    /**
+     * @notice Returns the underwriting pool asset for the given index.
+     * @param index_ The underwriting pool asset index.
+     * @return asset_ The underwriting pool asset.
+     * @return assetType_ The underwriting pool asset.
+    */
+    function assetAt(uint256 index_) external view override returns (address asset_, AssetType assetType_) {
+      address asset = _indexToAsset[index_];
+      return (asset, _assets[asset]);
+    }
+
+    /**
+     * @notice The maximum amount of cover that Solace as a whole can sell.
+     * @return cover The max amount of cover in wei.
+    */
+    function maxCover() external view override returns (uint256 cover) {
+      // get vault balance
+      address vault = _registry.get("vault");
```

Comment:
We can remove all "Vault" related stuff from data provider. The Vault will be removed and in the registry, if the  "vault" key is not found this code will raise error.

Reply:
Ok done, removed Vault from CoverageDataProvider.sol and CoverageDataProvider.test.ts